atlasdb-client/src/main/java/com/palantir/atlasdb/util/MetricsManager.java

```diff
@@ -62,6 +62,8 @@ private synchronized void registerMetricWithFqn(String fullyQualifiedMetricName,
         try {
             metricRegistry.register(fullyQualifiedMetricName, metric);
             registeredMetrics.add(fullyQualifiedMetricName);
+        } catch (IllegalArgumentException e) {
+            //Ignore metric already exists, to avoid big stacktraces.
```

Comment:
what specifically is triggering this case? I have a feeling this is due to gauges for a previously registered pool that was thrown away, and you need to actually remove any previously registered metrics with that name before adding the new one, otherwise you're going to leak memory and have the wrong metrics registered.